pkg/scheduling/volumeusage.go

```diff
@@ -77,38 +80,25 @@ func (u Volumes) Insert(volumes Volumes) {
 func GetVolumes(ctx context.Context, kubeClient client.Client, pod *v1.Pod) (Volumes, error) {
 	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("pod", pod.Name))
 	podPVCs := Volumes{}
-	defaultStorageClassName, err := DiscoverDefaultStorageClassName(ctx, kubeClient)
-	if err != nil {
-		return nil, fmt.Errorf("discovering default storage class, %w", err)
-	}
 	for _, volume := range pod.Spec.Volumes {
 		ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("volume", volume.Name))
-		var pvcID, storageClassName, volumeName string
-		var pvc v1.PersistentVolumeClaim
-		if volume.PersistentVolumeClaim != nil {
-			if err = kubeClient.Get(ctx, client.ObjectKey{Namespace: pod.Namespace, Name: volume.PersistentVolumeClaim.ClaimName}, &pvc); err != nil {
-				return nil, err
-			}
-			pvcID = fmt.Sprintf("%s/%s", pod.Namespace, volume.PersistentVolumeClaim.ClaimName)
-			storageClassName = lo.FromPtr(pvc.Spec.StorageClassName)
-			volumeName = pvc.Spec.VolumeName
-		} else if volume.Ephemeral != nil {
-			// generated name per https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#persistentvolumeclaim-naming
-			pvcID = fmt.Sprintf("%s/%s-%s", pod.Namespace, pod.Name, volume.Name)
-			storageClassName = lo.FromPtr(volume.Ephemeral.VolumeClaimTemplate.Spec.StorageClassName)
-			volumeName = volume.Ephemeral.VolumeClaimTemplate.Spec.VolumeName
-		} else {
-			continue
+		pvc, err := volumeutil.GetPersistentVolumeClaim(ctx, kubeClient, pod, volume)
+		if err != nil {
+			return nil, err
 		}
-		if storageClassName == "" {
-			storageClassName = defaultStorageClassName
+		// may not have pvc
+		if pvc == nil {
+			continue
 		}
+		volumeName := pvc.Spec.VolumeName
+		storageClassName := lo.FromPtr(pvc.Spec.StorageClassName)
 		driverName, err := resolveDriver(ctx, kubeClient, volumeName, storageClassName)
 		if err != nil {
 			return nil, err
 		}
 		// might be a non-CSI driver, something we don't currently handle
 		if driverName != "" {
+			pvcID := fmt.Sprintf("%s/%s", pod.Namespace, pvc.ObjectMeta.Name)
```

Comment:
```suggestion
			pvcID := fmt.Sprintf("%s/%s", pod.Namespace, pvc.Name)
```

No need to specify the `ObjectMeta` piece. Easier to just do `pvc.Name`